src/nspawn/nspawn-seccomp.c

```diff
@@ -187,7 +187,7 @@ int setup_seccomp(uint64_t cap_list_retain, char **syscall_whitelist, char **sys
                         return r;
 
                 r = seccomp_load(seccomp);
-                if (IN_SET(r, -EPERM, -EACCES))
+                if (IN_SET(r, -EPERM, -EACCES, -ENOMEM, -EINVAL, -EFAULT))
```

Comment:
hmm, could you add a wrapper macro or so to seccomp-util.h that makes this check? i.e. ERRNO_IS_SECCOMP_FATAL() or so, taking inspiration from ERRNO_IS_DISCONNECT() and similar in errno-util.h? And the reasons why these should be considered fatal that you included in the commit msg would be great to have as a comment hear that definition then

Reply:
Hmm, about EINVAL: I am pretty sure this is actually also generated in some cases where we have some syscall listed that hte kernel/libseccomp don't know, hence I think EINVAL should not be listed there...

Reply:
Ah ok. EINVAL I encompassed so many things it wasn't clear if it should have been fail open or fail close..